src/zos-mkdtemp.c

```diff
@@ -0,0 +1,105 @@
+//////////////////////////////////////////////////////////////////////////////
+//// Licensed Materials - Property of IBM
+//// ZOSLIB
+//// (C) Copyright IBM Corp. 2020. All Rights Reserved.
+//// US Government Users Restricted Rights - Use, duplication
+//// or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+/////////////////////////////////////////////////////////////////////////////////
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <sys/stat.h>
+
+typedef enum {
+  cp_ebcdic,
+  cp_ascii
+} CodePage;
+
+#define XXXXXX "XXXXXX"
+#define ValidChars "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_.,-"
+typedef struct CodePage {
+  char xes[sizeof(XXXXXX)];
+  char validChars[sizeof(ValidChars)];
+}  CodePage_t;
+
+#pragma convert("IBM-1047")
+static const CodePage_t CharTraitsEbcdic = {
+   XXXXXX,
+   ValidChars
+};
+#pragma convert(pop)
+
+#pragma convert("ISO8859-1")
+static const CodePage_t CharTraitsAscii = {
+   XXXXXX,
+   ValidChars
+};
+#pragma convert(pop)
+
+enum {
+  TempLen = sizeof(XXXXXX)-1
+};
+
+char *real_mkdtemp(CodePage_t cp, char *templ) {
+  if (!templ) {
```

Comment:
Suggest `if (!templ || !(p=strstr(templ, XXXXXX)) || memcmp(p, XXXXXX, sizeof(XXXXXX))) {`

Reply:
Just found out there's no strrstr() on z/OS.

Reply:
Sorry, just realized there's already a check for the above.

Reply:
np.  You would also need to check the length of templ before doing anything to the right of the first ||.  If templ was less than XXXXXX then the results would be undefined.